services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api/_core.py

```diff
@@ -274,20 +282,14 @@ async def are_all_services_present(
     """
     The dynamic-sidecar stack always expects to have 2 running services
     """
-    async with docker_client() as client:
-        stack_services = await client.services.list(
-            filters={
-                "label": [
-                    f"swarm_stack_name={dynamic_sidecar_settings.SWARM_STACK_NAME}",
-                    f"uuid={node_uuid}",
-                ]
-            }
-        )
-        if len(stack_services) != 2:
-            log.warning("Expected 2 services found %s", stack_services)
-            return False
+    stack_services = await _get_dynamic_sidecar_stack_services(
+        node_uuid, dynamic_sidecar_settings
+    )
+    if len(stack_services) != 2:
+        log.warning("Expected 2 services found %s", stack_services)
```

Comment:
THOUGHT: 

Moreover, these check functions might have three states: True, False and "don't know". 

For instance, ``is_dynamic_sidecar_stack_missing`` might fail the call to docker client and you cannot answer True or False. 

Now these functions raise an exception. How is this handled. Does the caller know about this third state? **I wonder** if it would be best for this type of functions to catch all exceptions and add an extra return state to "don't know" ...